### ISSUE TITLE
Discrepancies in parsing behavior of http requests

### ISSUE THREAD

**Issue by blessingcharles** [haproxy/haproxy#1728]
### Description

1. According to [RFC](https://datatracker.ietf.org/doc/html/rfc7230#section-2.6) , the http-name should be %x48.54.54.50 / "HTTP"[case-sensitive] , but haproxy accepts the following list of characters [0x2e , 0x2f ] + [0x30 - 0x39] + [0x48 , 0x50 , 0x52 , 0x53 , 0x54 ] in any characters of HTTP [H,T,T,P] == [. /] + [0-9] + [H , P , T , S , R]

#### RFC ABNF grammar

HTTP-version  = HTTP-name "/" DIGIT "." DIGIT
HTTP-name     = %x48.54.54.50 ; "HTTP", case-sensitive

### Expected Behavior

1. According to RFC for HTTP-name  should be "HTTP" , %x48.54.54.50 ans case-sensitive also , which is explicitly mentioned in the RFC 

### Steps to Reproduce the Behavior

1. Accepts [0x2e , 0x2f ] + [0x30 - 0x39] + [0x48 , 0x50 , 0x52 , 0x53 , 0x54 ] 
    in any characters of HTTP [H,T,T,P] == [. /] + [0-9] +[H , P , T , S , R]

```bash
echo -ne "GET / ..T./3.1\r\nContent-Length: 3\r\n\r\naaa" | nc localhost 9003

echo -ne "GET / ././/3.1\r\nContent-Length: 3\r\n\r\naaa" | nc localhost 9003

echo -ne "GET / .0.//3.1\r\nContent-Length: 3\r\n\r\naaa" | nc localhost 9003

echo -ne "GET / .H../3.1\r\nContent-Length: 3\r\n\r\naaa" | nc localhost 9003
``` 

### What is your configuration?

```haproxy
- haproxy version : HAProxy version 2.5.7-2ef551d 2022/05/13 

## Dockerfile

FROM haproxy
COPY haproxy.cfg /usr/local/etc/haproxy/haproxy.cfg

## Haproxy Config

global
    daemon
    maxconn 256

defaults
    mode http
    timeout connect 5000ms
    timeout client 50000ms
    timeout server 50000ms

listen http-in
    bind *:80
    server server1 172.17.0.1:5000 maxconn 32
```

### Output of `haproxy -vv`

```plain
### Docker interactive mode output

[NOTICE]   (1) : New worker (8) forked
[NOTICE]   (1) : Loading success.
```

**Comment by capflam** [comment]
It was recently fixed: 8f4116ea

**Comment by capflam** [comment]
Backported as far as 2.0. I'm closing the issue now. Thanks !
